packages/styles/src/components/header/_mixins.scss

```
@@ -32,6 +32,7 @@
   }
 
   &.selected,
+  &:where(a)[aria-current='page'],
```

You're not restricting this on desktop, to be consistent let's not restrict it here as well: 
```suggestion
  &[aria-current='page'],
```